parlai/core/logs.py

```diff
@@ -234,6 +246,17 @@ def log_final(self, setting, report):
         }
         for key, value in report.items():
             self.run.summary[key] = value
+        if self.model_file is not None and self.log_model:
+            artifact = self._wandb.Artifact('model', type='model')
+            artifact.add_file(
+                self.model_file,
+                f'{self.run.name}_model',
+            )
+            artifact.add_file(
+                self.model_file + ".dict",
```

Comment:
Assuming `f"{self.model_file}.dict"` is available when initializing this logger, I think we should log it in the `init` itself instead of logging it every time. If it's updating every time, we can just resolve this comment.

So a model artifact has three components - model, dict, and vocab. Will the user have to download the same separately? A utility function to download the correct model and associated dict and vocab provided URI like `parambharat/parlai/daily-wave-8-model_tr4:v3` should be useful.

Reply:
Since the file checksum doesn't change it only logs one artifact. Verified in the UI